### ISSUE TITLE
winget CLI crashes if I try to install, show, search, list, upgrade, uninstall, export

### ISSUE THREAD

**Issue by mywave82** [microsoft/winget-cli#3922]
### Brief description of your issue

Faulting application name: winget.exe, version: 1.17.2203.10001, time stamp: 0x622a8652
Faulting module name: WindowsPackageManager.dll, version: 1.17.2203.10001, time stamp: 0x622a8613
Exception code: 0xc0000005
Fault offset: 0x0000000000089bf3
Faulting process id: 0x0x2A4C
Faulting application start time: 0x0x1DA1DF86B8EFFE4
Faulting application path: C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_1.17.10691.0_x64__8wekyb3d8bbwe\winget.exe
Faulting module path: C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_1.17.10691.0_x64__8wekyb3d8bbwe\WindowsPackageManager.dll
Report Id: c4ef8304-6816-4d4d-ab91-cced96aa25c0
Faulting package full name: Microsoft.DesktopAppInstaller_1.17.10691.0_x64__8wekyb3d8bbwe
Faulting package-relative application ID: winget

[-](https://github.com/microsoft/winget-cli/issues/new?assignees=&labels=&projects=&template=Bug_Report.yml#) <Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
[-](https://github.com/microsoft/winget-cli/issues/new?assignees=&labels=&projects=&template=Bug_Report.yml#) <System>
  <Provider Name="Application Error" Guid="{a0e9b465-b939-57d7-b27d-95d8e925ff57}" /> 
  <EventID>1000</EventID> 
  <Version>0</Version> 
  <Level>2</Level> 
  <Task>100</Task> 
  <Opcode>0</Opcode> 
  <Keywords>0x8000000000000000</Keywords> 
  <TimeCreated SystemTime="2023-11-23T10:32:55.8344100Z" /> 
  <EventRecordID>4073</EventRecordID> 
  <Correlation /> 
  <Execution ProcessID="29260" ThreadID="26104" /> 
  <Channel>Application</Channel> 
  <Computer>Stians-Laptop</Computer> 
  <Security UserID="S-1-12-1-830122842-1118079056-2203915663-2701775520" /> 
  </System>
[-](https://github.com/microsoft/winget-cli/issues/new?assignees=&labels=&projects=&template=Bug_Report.yml#) <EventData>
  <Data Name="AppName">winget.exe</Data> 
  <Data Name="AppVersion">1.17.2203.10001</Data> 
  <Data Name="AppTimeStamp">622a8652</Data> 
  <Data Name="ModuleName">WindowsPackageManager.dll</Data> 
  <Data Name="ModuleVersion">1.17.2203.10001</Data> 
  <Data Name="ModuleTimeStamp">622a8613</Data> 
  <Data Name="ExceptionCode">c0000005</Data> 
  <Data Name="FaultingOffset">0000000000089bf3</Data> 
  <Data Name="ProcessId">0x2a4c</Data> 
  <Data Name="ProcessCreationTime">0x1da1df86b8effe4</Data> 
  <Data Name="AppPath">C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_1.17.10691.0_x64__8wekyb3d8bbwe\winget.exe</Data> 
  <Data Name="ModulePath">C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_1.17.10691.0_x64__8wekyb3d8bbwe\WindowsPackageManager.dll</Data> 
  <Data Name="IntegratorReportId">c4ef8304-6816-4d4d-ab91-cced96aa25c0</Data> 
  <Data Name="PackageFullName">Microsoft.DesktopAppInstaller_1.17.10691.0_x64__8wekyb3d8bbwe</Data> 
  <Data Name="PackageRelativeAppId">winget</Data> 
  </EventData>
  </Event>

### Steps to reproduce

winget upgrade

### Expected behavior

That it tries to upgrade my programs

### Actual behavior

Immidiatly crashes (it starts to print the spinner, and exists back to the shell)

### Environment

```shell
C:\Users\StianSkjelstad\Downloads>winget --info
Windows Package Manager v1.2.10691
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22621.2715
Package: Microsoft.DesktopAppInstaller v1.17.10691.0

Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale


Out of the box, fresh laptop, joined azure ad, "Microsoft 365 Buisness Standard", and did perform windows update. "Windows Store" icon also went missing.
```

**Comment by Trenly** [comment]
You’re using an extremely old version of winget and the CDN for that version has been deprecated. I suggest waiting for the store to automatically update Winget or manually updating it yourself

**Comment by mywave82** [comment]
I am probably stuck with that old version, since windows store does not open. Doing wsreset.exe does this:

![image](https://github.com/microsoft/winget-cli/assets/12225220/d2b494c5-145b-4c4b-83fa-e3b6f2480d94)

I will try to do full reset of windows, and this time start windows store and winget before doing windows update.

**Comment by mywave82** [comment]
I performed "Reset Windows" and used winget before doing windows update. It now works as expected and I now have "App Installer" in the list of installed programs. So this means that if you do a windows update before using winget on a fresh Windows installatation, you might end up stuck with a broken version?

The reason for Windows Store missing is likely that the computer is setup with a work account "Microsoft 365 Business Standard" so it missing Intune etc.

**Comment by mywave82** [comment]
> FYI, an easy way to update winget without using the Store is to do `Add-AppPackage https://aka.ms/GetWinGet`. The aka.ms link points to the latest msixbundle available for download from here on GitHub.

So there is a work-around :-) But still out-of-the-box installation of Windows-11 should just work, so something needs to change in what Windows Update installs since the current experience is broken and all howto's you find searching online does not work.

**Comment by denelon** [comment]
This was related to CDN issues and has been resolved in WinGet 1.7.
